link_test.go

```diff
@@ -664,3 +667,151 @@ func TestLinkStats64UnmarshalBinary(t *testing.T) {
 		})
 	}
 }
+
+func TestLinkXDP(t *testing.T) {
+	// BPF compilation requires a high RLIMIT_MEMLOCK
+	n := uint64(1024 * 1024 * 10)
+	err := unix.Setrlimit(unix.RLIMIT_MEMLOCK, &unix.Rlimit{Cur: n, Max: n})
+	if err != nil {
+		t.Fatalf("failed to increase RLIMIT_MEMLOCK: %s", err)
+	}
+
+	// establish a netlink connections
+	conn, err := Dial(nil)
+	if err != nil {
+		t.Fatalf("failed to establish netlink socket")
+	}
+	defer conn.Close()
+
+	// construct dummy interface to test XDP program against
+	if err := conn.Link.New(&LinkMessage{
+		Family: unix.AF_UNSPEC,
+		Index:  1001,
+		Flags:  unix.IFF_UP,
+		Attributes: &LinkAttributes{
+			Name: "dummy0",
+			Info: &LinkInfo{Kind: "dummy"},
+		},
+	}); err != nil {
+		t.Fatalf("failed to construct dummy interface: %s", err)
+	}
+
+	// get info for the dummy interface
+	interf, err := conn.Link.Get(1001)
+	if err != nil {
+		conn.Link.Delete(interf.Index)
+		t.Fatalf("could not get dummy interface info: %s", err)
+	}
+
+	// compile a BPF test program. If it fails error out of the tests
+	// and clean up dummy interface.
+	prog, err := ebpf.NewProgram(&ebpf.ProgramSpec{
+		Type: ebpf.XDP,
+		Instructions: asm.Instructions{
+			asm.LoadImm(asm.R0, int64(2), asm.DWord),
+			asm.Return(),
+		},
+		License: "MIT",
+	})
+	if err != nil {
+		conn.Link.Delete(interf.Index)
+		t.Fatalf("failed to compile test bpf program: %s", err)
+	}
+
+	prog2, err := ebpf.NewProgram(&ebpf.ProgramSpec{
```

Comment:
You could define an instance of `ebpf.ProgramSpec` once above and refer to it in your calls to `ebpf.NewProgram()`. Saves loc and less likely to have implementation errors.